repoman/git/depot_operations.py

```diff
@@ -42,56 +42,39 @@ def check_changeset_availability(self, path, changesets):
         """
         Inherited method :func:`~DepotOperations.check_changeset_availability`
         """
+        # TODO: Check if changesets are really in path
         return changesets
 
     def grab_changesets(self, path, url, changesets):
         """
         Inherited method :func:`~DepotOperations.grab_changesets`
         """
         logger.debug('Grabbing changesets from %s to %s' % (url, path))
-        git_repo = pygit2.Repository(path)
-        try:
-            origin = self._set_origin_source(git_repo, url)
-            result = origin.fetch()
-            logger.debug('GIT Done grabbing changesets (%s)' % (result))
-            self._save_state(path)
-        except pygit2.GitError as e:
-            logger.exception('Error Grabbing changesets: %s' % e)
-            return False
+        sh.git("fetch", url, "+refs/heads/*:refs/remotes/origin/*", _cwd=path)
+        self._save_state(path)
         return True
 
     def init_depot(self, path, parent=None, source=None):
         """
         Inherited method :func:`~DepotOperations.init_depot`
         """
-        try:
-            logger.info('Initializing Depot %s with parent %s' % (
-                        path, parent))
-            bare = not parent
+        logger.info('Initializing Depot %s with parent %s' % (
+                    path, parent))
+ 
+        sh.git('init', path, bare=not parent)
```

Comment:
Should we use `git clone`? 
In that case, remote default branch is retrieved. There are repos whose default branch is different from master.

Reply:
I started implementing this with clone, yes, but this method (I think that according to tests) is intended to create an empty repository.
I don't think this method should handle the default branch, maybe in grab_changeset it has to be taken into account.
Also I don't think that callers of these methods should assume that an specific branch is checked out, they should explicitly update to this branch.